core/registration.c

```diff
@@ -1330,16 +1396,21 @@ void registration_step(lwm2m_context_t * contextP,
             }
         }
         break;
-
+        case STATE_REG_PARTIAL_DONE:
+            prv_register(contextP, targetP);
```

Comment:
In the example client, this function may only get called every 60s. That seems a long time to wait between messages, and would place an upper limit of 4 blocks withing the EXCHANGE_LIFETIME. Can sending the next block be triggered by receipt of the response instead?

Reply:
One alternative might be to set *timeoutP to 1 (or 2, the CoAP ACK_TIMEOUT) while the blocks are being sent. Battery powered devices may have tradeoffs between battery life, the frequency of checking, and the total time they need to have a radio powered on. Triggering the next send on receipt of the response would likely be best for battery life.

Reply:
Sorry for delayed response, As per my understanding in order to trigger next block transfer we need to call prv_register() for which we need a pointer to lwm2m_context, which is not there in handle_RegistrationReply(). Is there any implementation which comes to your mind ?

Reply:
I hadn't considered that, I'm actually using a modified version that does pass in the context. Modifying the transaction callback to pass the context is one option, but isn't backward compatible.

If transaction->userData is set to the context instead of the server, the server could be determined by looking for a server with a sessionH matching transaction->peerH (use lwm2m_session_is_equal).

Reply:
There is a utils_findServer function, in utils.c, I think it does the same as what you mentioned.